### DISCUSSION TITLE
Turtle module compatibility

### DISCUSSION THREAD

**Discussion by jowe19** [tndrle/PyLuaTeX#16] [General]
Hi, 
I found an example for using `Matplotlib` together with `pyluatex`. Does there exists a way to print the result of a turtle figure ?

**Comment by tndrle** [comment]
Hi,
unfortunately, I don't know what a turtle figure is. Can you give an example (code) and also what problems you have with it?
Cheers

**Comment by jowe19** [comment]
Please find below a small example. 

```
\documentclass{article}

\usepackage{pyluatex}
A

\begin{pythonq}
from turtle import *

resetscreen() # efface l'écran
up() # arrête le tracé
setposition(-200,0) # se place à gauche de la fenêtre
down() # remets le tracé

# tracé de la courbe en profondeur 1 pour une longueur 400
l = 400
forward(l/3) # en avant
left(60)
forward(l/3) # en avant
right(120)
forward(l/3) # en avant
left(60)
forward(l/3) # en avant
\end{pythonq}

\end{document}
```

The typesetting doesn't finish, I think.

**Comment by tndrle** [comment]
Thanks for the example. As far as I understand, Turtle creates a Tkinter canvas for drawing, which is shown in its own window. There is no way for LaTeX to know that the Python code creates an image which you want to include in the document.
Maybe you can save the image as a file and include that, e.g. with `turtle.getscreen().getcanvas().postscript(file='image.eps')`.
However, I think LuaLaTeX cannot include PostScript files, so maybe also try the [`epstopdf`](https://www.ctan.org/pkg/epstopdf) LaTeX package.

The code could look something like this (not tested)
```latex
\documentclass{article}

\usepackage{pyluatex}
\usepackage{graphicx}
\usepackage{epstopdf}

\begin{document}
\begin{pythonq}
import turtle

... your code ...

turtle.getscreen().getcanvas().postscript(file='image.eps')
\end{pythonq}
\includegraphics{image}
\end{document}
```

**Comment by jowe19** [comment]
Thanks for the hint, but the typesetting doesn't finish with the following code. 

```
\documentclass{article}

\usepackage{pyluatex}
\usepackage{graphicx}
\usepackage{epstopdf}
A

\begin{pythonq}
from turtle import *

resetscreen() # efface l'écran
up() # arrête le tracé
setposition(-200,0) # se place à gauche de la fenêtre
down() # remets le tracé

# tracé de la courbe en profondeur 1 pour une longueur 400
l = 400
forward(l/3) # en avant
left(60)
forward(l/3) # en avant
right(120)
forward(l/3) # en avant
left(60)
forward(l/3) # en avant
turtle.getscreen().getcanvas().postscript(file='image.eps')
\end{pythonq}
\includegraphics{image}

\end{document}
```

**Reply by tndrle** [reply]
Well, most importantly, `\begin{document}` is missing. Also, there is a lonely `A` which might cause problems. And in the last line of Python code you have to remove `tutle.` (since you import all from turtle with `from turtle import *` instead of doing `import turtle`).

If you correct these points, the document should be compilable. Right now it is not.

Then, you still might get problems with `epstopdf` because it must be installed correctly (it calls a command line tool to do the conversion to PDF). This also depends on your system, so I cannot really give advice here.
